src/ReaderScreen/components/AudioPlayer/hooks/useAudioSyncScroll.js

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useRef } from "react";
+import { useSelector } from "react-redux";
+import satnamSinghBaniTimestamps from "@service/2_SatnamSingh";
+import baniTimestamps from "@service/bani_timestamps";
+
+const useAudioSyncScroll = (progress, isPlaying, webViewRef, baniID, artistID) => {
+  const isAudioSyncScroll = useSelector((state) => state.isAudioSyncScroll);
+  const lastSequenceRef = useRef(null);
+  const isScrollingRef = useRef(false);
+
+  // Find current sequence based on audio progress
+  const findCurrentSequence = (currentTime, artist) => {
+    if (!currentTime || currentTime === 0) return null;
+
+    // Find the timestamp entry that contains the current time
+    const baniLRC = artist === 4 ? baniTimestamps : satnamSinghBaniTimestamps;
+    const currentTimestamp = baniLRC.find(
+      (timestamp) => currentTime >= timestamp.start && currentTime <= timestamp.end
+    );
+
+    return currentTimestamp ? currentTimestamp.sequence : null;
+  };
+
+  // Scroll to specific sequence in WebView
+  const scrollToSequence = (sequence) => {
+    if (!webViewRef?.current?.postMessage || !sequence) {
+      return;
+    }
+
+    try {
+      // Prevent multiple rapid scroll calls
+      if (isScrollingRef.current) return;
+      isScrollingRef.current = true;
+
+      const scrollMessage = {
+        action: "scrollToSequence",
+        sequence,
+        behavior: "smooth", // smooth scrolling
+      };
+
+      webViewRef.current.postMessage(JSON.stringify(scrollMessage));
+
+      // Reset scrolling flag after a short delay
+      setTimeout(() => {
```

Comment:
This setTimeout doesn't have cleanup, potentially causing memory leaks.

Reply:
fixed